server/src/main/java/org/elasticsearch/search/aggregations/AggregationBuilder.java

```diff
@@ -219,6 +219,18 @@ public boolean isInSortOrderExecutionRequired() {
         return false;
     }
 
+    /**
+     * Return false if this aggregation or any of the child aggregations does not support concurrent search
+     */
+    public boolean supportsConcurrentExecution() {
+        for (AggregationBuilder builder : factoriesBuilder.getAggregatorFactories()) {
+            if (builder.supportsConcurrentExecution() == false) {
+                return false;
+            }
+        }
+        return isInSortOrderExecutionRequired() == false;
```

Comment:
Interesting. How many aggs don't support concurrent execution due to this last conditional?

Reply:
And that one never gets executed in concurrent mode because time series aggregation had their own executor.